controllers/broker_controller.go

```diff
@@ -79,18 +82,24 @@ func (r *BrokerReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 	// Get broker endpoint
 	err = brokerpackage.GetEndpoint(ctx, baseResource, r.Clientset, r.KubeConfig.Host)
 	if err != nil {
-		return ctrl.Result{}, ErrGetEndpoint(err)
+		err = ErrGetEndpoint(err)
+		r.Log.Error(err, "unable to get the broker endpoint")
+		return ctrl.Result{}, err
 	}
 
 	// Patch the broker resource
 	patch, err := utils.Marshal(baseResource)
 	if err != nil {
-		return ctrl.Result{}, ErrUpdateResource(err)
+		err = ErrUpdateResource(err)
+		r.Log.Error(err, "unable to update broker resource")
+		return ctrl.Result{}, err
 	}
 
 	err = r.Status().Patch(ctx, baseResource, client.RawPatch(types.MergePatchType, []byte(patch)))
 	if err != nil {
-		return ctrl.Result{}, ErrUpdateResource(err)
+		err = ErrUpdateResource(err)
+		r.Log.Error(err, "unable to update broker resource")
```

Comment:
Meaning that the configuration was rejected from Kube API or that the resource isn't present or that the requested configuration was missing or invalid or what?